tests/test_read_input_file.py

```diff
@@ -0,0 +1,52 @@
+import sys
+
+from pathlib import Path
+
+import pytest
+import unittest
+
+from src.read_input_file import read_input_file
+from src.package import PackageMetadata
+
+def test_assertion_when_package_has_no_arch():
+    """Input file with entries that don't have architecture, should fail."""
+    test_case = unittest.TestCase()
+    with test_case.assertRaises(ValueError) as ctx:
+        read_input_file(
+            registry_path=Path("../resources/registry"), 
+            input_file=Path("../_main/tests/resources/faulty_input.in").resolve()
+        )
+    
+    test_case.assertEqual(
+        str(ctx.exception), 
+        "Entry has unexpected format, expected format: name:arch=version or name:arch"
+    )
```

Comment:
I would do this with pytest instead:
```suggestion
    with pytest.raises(ValueError, match="Entry has unexpected format, expected format: name:arch=version or name:arch"):
        read_input_file(
            registry_path=Path("../resources/registry"), 
            input_file=Path("../_main/tests/resources/faulty_input.in").resolve()
        )
```
I'm also a big fan of custom exception types (i.e. subclassing RuntimeError or ValueError), then you can be very specific about what will/should be raised and the user can be explicit about what to catch and handle